ports/pulsar-client-cpp/unofficial-pulsar-config.cmake

```
@@ -0,0 +1,37 @@
+if (NOT TARGET unofficial::pulsar::pulsar)
+    get_filename_component(VCPKG_IMPORT_PREFIX "${CMAKE_CURRENT_LIST_FILE}" PATH)
+    get_filename_component(VCPKG_IMPORT_PREFIX "${VCPKG_IMPORT_PREFIX}" PATH)
+    get_filename_component(VCPKG_IMPORT_PREFIX "${VCPKG_IMPORT_PREFIX}" PATH)
+
+    find_path(_pulsar_include_dir NAMES "pulsar/Client.h" PATH ${VCPKG_IMPORT_PREFIX})
+    if (CMAKE_BUILD_TYPE STREQUAL "Debug")
+        set(VCPKG_IMPORT_PREFIX "${VCPKG_IMPORT_PREFIX}/debug")
+    endif ()
+    if ("@VCPKG_LIBRARY_LINKAGE" STREQUAL "dynamic" OR PULSAR_FORCE_DYNAMIC_LIBRARY)
+        find_library(_pulsar_library NAMES libpulsar.so libpulsar.dylib pulsar.lib PATH ${VCPKG_IMPORT_PREFIX})
+    else ()
+        find_library(_pulsar_library NAMES libpulsar.a pulsar-static.lib PATH ${VCPKG_IMPORT_PREFIX})
+    endif ()
+    if (NOT _pulsar_include_dir OR NOT _pulsar_library)
+        message(FATAL_ERROR "Broken installation of vcpkg port pulsar-client-cpp")
+    endif ()
+
+    find_package(protobuf CONFIG REQUIRED)
+    find_package(CURL CONFIG REQUIRED)
+    find_package(zstd CONFIG REQUIRED)
+    find_package(snappy CONFIG REQUIRED)
```

```suggestion
    include(CMakeFindDependencyMacro)
    find_dependency(protobuf)
    find_dependency(CURL)
    find_dependency(zstd)
    find_dependency(snappy)
```

This should use `find_dependency` instead